### PR TITLE
stm32l4-multi: Reorganize config, disable defaults

### PR DESCRIPTION
**Pull request by anglov**
Peripherial defaults removed from config.h - project shallset required peripheria explicitly in board_config.h.

Added error when UART_CONSOLE is not set and warning on console setted to disabled UART. Moved errors to .c file to not reporting it many times.

Reduce config.h variable propagation by removing unnecessary includes and move computed variables to valid .c file.

JIRA: DEND-32

## Motivation and Context

Better control of enabled/disabled peripheria via board_config
Small clean up.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

Need to define missing variables in case that project rely on that

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: armv7m4-stm32l4x6-nucleo

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: multi/stm32l4-multi/stm32l4-multi.c [hunk 1]
```diff
@@ -41,6 +42,30 @@
...
+#ifndef UART_CONSOLE
+#error "UART_CONSOLE not specified"
+#endif
+
+#if !(UART_CONSOLE == 1 && (UART1 || TTY1)) && \
+	!(UART_CONSOLE == 2 && (UART2 || TTY2)) && \
+	!(UART_CONSOLE == 3 && (UART3 || TTY3)) && \
+	!(UART_CONSOLE == 4 && (UART4 || TTY4)) && \
+	!(UART_CONSOLE == 5 && (UART5 || TTY5))
+#warning "Console enabled on disabled UART/TTY"
```
Comment (anglov) [inline multi/stm32l4-multi/stm32l4-multi.c:59]: Thinked about it if this should be treated as warning or error
In fact this is currently the only way to fastly silence the console - set it to invalid UART.
Decided to warning, but I'm open to suggestion

File: multi/stm32l4-multi/config.h [hunk 2]
```diff
@@ -46,24 +49,19 @@
...
 
 #ifndef TTY5
 #define TTY5 0
 #endif
 
-#if (UART1 && TTY1) || (UART2 && TTY2) || (UART3 && TTY3) || \
-	(UART4 && TTY4) || (UART5 && TTY5)
-#error "Can't use UART as UART and TTY at the same time!"
-#endif
+/* UART_CONSOLE have to be specified in board_config.h */
```
Comment (agkaminski) [inline multi/stm32l4-multi/config.h:59]:
```suggestion
/* UART_CONSOLE has to be specified in board_config.h */
```

**Review by agkaminski — APPROVED** [review]
LGTM

**Comment by anglov** [comment]
Fixed typo and accidentally removed blank line

**Review by agkaminski — APPROVED** [review]

**Pull request merged**